Source/Core/VideoCommon/TextureCacheBase.cpp

```diff
@@ -138,7 +144,11 @@ void TextureCacheBase::Cleanup(int _frameCount)
   TexAddrCache::iterator tcend = textures_by_address.end();
   while (iter != tcend)
   {
-    if (iter->second->frameCount == FRAMECOUNT_INVALID)
+    if (iter->second->tmem_only)
+    {
+      iter = InvalidateTexture(iter);
```

Comment:
Cleanup is called at a random position within the frame (at least with XFB emulation), so this might flush your tmem cache a bit too early. What do you think about delaying this invalidation for one frame? It should be straight forward to set frameCount to last_frame_count + 1 - TEXTURE_KILL_THRESHOLD within the invalidation code.